tests/codeception/functional/Install/Content/VideoEmbedCest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Class VideoEmbedCest.
+ *
+ * @group install
+ */
+class VideoEmbedCest {
+
+  /**
+   * I can create a page with a video embed and verify caption.
+   */
+  public function testVideoEmbed(FunctionalTester $I) {
+    // Login and add flexible page
+    $I->logInWithRole('contributor');
+    $I->amOnPage('node/add/hs_basic_page');
+    $I->fillField('Title', 'Demo Basic Page');
+
+    // Add text field
+    $I->scrollTo('#edit-field-hs-page-components-add-more');
+    $I->click('List additional actions', '#edit-field-hs-page-components-add-more');
+    $I->scrollTo('#field-hs-page-components-hs-text-area-add-more');
+    $I->click('#field-hs-page-components-hs-text-area-add-more');
+    $I->wait(2);
+
+    // Add media
+    $I->click('.cke_button__drupalmedialibrary');
+    $I->waitForText('Add or select media');
+    $I->click('.media-library-menu__link[data-title="Video"]');
+    $I->waitForText('Add Video via URL');
+    $I->fillField('Add Video via URL', 'https://www.youtube.com/watch?v=95N_spFNEkY');
+    $I->click('.js-form-submit[value="Add"]');
+    $I->executeJS('document.querySelector("body").removeChild(document.querySelector(".ui-widget-overlay"))', []);
+    $I->waitForText('Save and select');
+    $I->executeJS('document.querySelector(".ui-dialog-buttonpane .button--primary").click()', []);
+    $I->waitForText('Insert selected');
+    $I->executeJS('document.querySelector("button.media-library-select").click()', []);
+    $I->wait(2);
+
+    // Enable caption
+    $I->switchToIFrame(".cke_wysiwyg_frame");
+    $I->executeJS('document.querySelector(".media-library-item__edit").style.display = "inline"', []);
+    $I->executeJS('document.querySelector(".media-library-item__edit").click()', []);
+    $I->switchToIFrame();
+    $I->waitForText('Caption');
+    $I->executeJS('document.querySelector("input[name=\"hasCaption\"]").click()', []);
+    $I->executeJS('document.querySelector(".ui-dialog-buttonset .js-form-submit").click()', []);
+    $I->wait(1);
```

Comment:
Use click actions to choose and select the options in the CKEditor

Reply:
This one is really tricky. The edit button for the video is only available on hover. And if I use ```$I->moveMouseOver('.cke_widget_element')``` or ```$I->moveMouseOver('.hb-media-video')``` it doesn't cause the button to show. So, I had to use JS to display the button. Secondarily, I tried to click the button after it was revealed, but none of the standard ```$I->click('Edit media') worked, so, again, I used JS to click it. If you've run into this before and have a better solution, I'm happy to hear one. I wonder if this is because it is inside an iFrame?

Reply:
Hmm. I don't think I have any examples of editing the media inside the WYSIWYG. we can leave this for now, but maybe there's a better way.